src/os/bluestore/BlueStore.cc

```diff
@@ -4819,15 +4838,16 @@ void BlueStore::_txc_finish_kv(TransContext *txc)
     txc->onreadable_sync = NULL;
   }
   if (txc->onreadable) {
-    finisher.queue(txc->onreadable);
+    finishers[txc->osr->id % m_finisher_num]->queue(txc->onreadable);
     txc->onreadable = NULL;
   }
   if (txc->oncommit) {
-    finisher.queue(txc->oncommit);
+    finishers[txc->osr->id % m_finisher_num]->queue(txc->oncommit);
     txc->oncommit = NULL;
   }
   while (!txc->oncommits.empty()) {
-    finisher.queue(txc->oncommits.front());
+    auto f = txc->oncommits.front();
+    finishers[txc->osr->id % m_finisher_num]->queue(f);
```

Comment:
I think the way to do this is not to use random ids, but to instead assign a shard id to the sequencer when it is created.  This can then map to the same mapping scheme that is distributing pg's across teh op_wq shards in the OSD and the collections across cache shards in BlueStore. Probably we should add the shard_hint to the ObjectStore::Sequencer class and initialize it in the OSD/PG code when the sequencer is created.  Elsewhere we use this to do the mapping:

 cid.hash_to_shard(num_shards)

So maybe make the config option a bool, and then use the g_conf->osd_op_num_shards config value that also drives op_wq shard and cache shard counts?